src/OpenTracing.AspNetCore/AspNetCoreFormatExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using Microsoft.AspNetCore.Http;
+using OpenTracing.AspNetCore;
+using OpenTracing.Propagation;
+
+namespace OpenTracing
+{
+    public static class AspNetCoreFormatExtensions
+    {
+        public static ISpanContext ExtractHttpHeaders(this ITracer tracer, HttpContext context)
```

Comment:
it's a standard in the ASP.NET Core framework. this is one of the reasons why I built this PoC, it shows the great potential of extension methods with `Inject`/`Extract` - framework implementations can provide their extension for whatever format is appropriate and users don't need to worry about wrapping the actual carrier into a opentracing-carrier.